content/en/docs/tasks/administer-cluster/configure-upgrade-etcd.md

```diff
@@ -200,6 +200,8 @@ If the access URLs of the restored cluster is changed from the previous cluster,
 
 If the majority of etcd members have permanently failed, the etcd cluster is considered failed. In this scenario, Kubernetes cannot make any changes to its current state. Although the scheduled pods might continue to run, no new pods can be scheduled. In such cases, recover the etcd cluster and potentially reconfigure Kubernetes API server to fix the issue.
 
+You should restart all Kubernetes system components after restoring the etcd cluster.
```

Comment:
This sounds important, so:
```suggestion
{{< note >}}
You should restart all Kubernetes control plane components after restoring the
etcd cluster, so as to be confident that the cluster is not relying on stale cache data.
{{< /note >}}
```
(do you also need to restart all nodes? I'm assuming not)

Reply:
kubelet also watches the apiserver and keeps some local cache I think, so I would suggest restarting them just to be safe. @jingyih do you have any experience restoring etcd in Kubernetes? If so, could you share your opinion on this?

Reply:
In general, we don't recommend restoring etcd when any apiserver is running. So what we recommend is:
- stop *all* kube-apiserver
- restore state in etcd
- restart all kube-apiserver

Given that the restore takes a bit of time, the critical components will loose leader lock and they will restart themselves (we enable leader election no matter how many of them are running by-default).
We should recommend restarting *any* components to ensure that they don't rely on some stale data, but in practice they will relist on their own after all kube-apiservers will be done for some time.

Reply:
Totally agree with @wojtek-t 's comment here. Restoring etcd cluster with multiple etcd servers is an offline process (i.e., cannot be performed in rolling fashion). It is better to stop the apiservers first.

Reply:
ack. Will update soon. Thanks for the suggestions!

Reply:
Sorry I lost track of this PR. I updated the paragraph accordingly. Please take a look

Reply:
I don't want to hijack the thread, but based on the changes made, should we consider using the `{{ caution }}` tag here? (where do we draw the line between `{{ note }}` and `{{ caution }}`?)

Reply:
Possibly update from note to caution (follow on PR)? The text changes look good. Thanks!